FEXCore/Source/Interface/Core/JIT/JIT.cpp

```diff
@@ -483,20 +483,16 @@ void Arm64JITCore::Op_Unhandled(const IR::IROp_Header* IROp, IR::NodeID Node) {
 }
 
 static void DirectBlockDelinker(FEXCore::Core::CpuStateFrame* Frame, FEXCore::Context::ExitFunctionLinkData* Record) {
-  auto LinkerAddress = Frame->Pointers.Common.ExitFunctionLinker;
-  uintptr_t branch = (uintptr_t)(Record)-8;
-  ARMEmitter::Emitter emit((uint8_t*)(branch), 8);
-  ARMEmitter::ForwardLabel l_BranchHost;
-  emit.ldr(TMP1, &l_BranchHost);
+  // Emit new 16 bytes of code to a temporary patch, then atomically apply it
+  __uint128_t Patch;
+  ARMEmitter::Emitter emit((uint8_t*)&Patch, sizeof(Patch));
+  emit.ldr(TMP1, 8); // PC-relative value pointing to constant after blr
   emit.blr(TMP1);
-  emit.Bind(&l_BranchHost);
-  emit.dc64(LinkerAddress);
-  ARMEmitter::Emitter::ClearICache((void*)branch, 8);
-}
+  emit.dc64(Frame->Pointers.Common.ExitFunctionLinker);
 
-static void IndirectBlockDelinker(FEXCore::Core::CpuStateFrame* Frame, FEXCore::Context::ExitFunctionLinkData* Record) {
-  auto LinkerAddress = Frame->Pointers.Common.ExitFunctionLinker;
-  Record->HostBranch = LinkerAddress;
+  auto branch = reinterpret_cast<__uint128_t*>((uintptr_t)Record - 8);
+  std::atomic_ref<__uint128_t>(*branch).store(Patch, std::memory_order::relaxed);
```

Comment:
Sorry, this actually needs to be an ldaxp+stlxp+b.cc loop to ensure single-copy atomicity of a 128-bit aligned region. I just reread the spec on LSE2 around single-copy atomicity and LDP/STP instructions are actually considering two atomic accesses.

Reply:
You've done deeper research than me, but are you sure? Wouldn't this be a bug in the `atomic_ref` implementation?

Reply:
Perhaps all references I've been looking at are exclusive to Armv8.4, and it's LSE2+v8.2 where what you're saying applies?

Reply:
Oh right, you're right. I didn't read far enough in to the spec.